### DISCUSSION TITLE
Raspberry Pi Pico - First project problem

### DISCUSSION THREAD

**Discussion by ThiNguyen3008** [micropython/micropython#10369] [General]
Hello, I'm a beginner for Raspberry Pi Pico. I cannot run my first project and I don't know why. I'm using Thonny 4.0.1. Can you help me please? Thank you
This is from the book
![as](https://user-images.githubusercontent.com/114266186/210022879-9c76600d-99f2-4660-aa9a-d846f888eafd.JPG)
This is a screenshot of my code
![ask](https://user-images.githubusercontent.com/114266186/210022886-d77aab0c-f5fe-47d4-ac72-ba7abc702be4.JPG)

**Comment by beetlegigg** [comment]
The statements in your while loop need to be indented.  

```
while True:
    led_onboard.value(1)
    utime.sleep(2)
    etc....
```

I suggest you will save yourself some hair pulling if you just take a small pause to introduce yourself to python (and micropython) .  I was perusing youtube before looking at this forum and I remembered youtube showing me some latest videos one of which was a new series on learning micropython  so I popped back to get the address of lesson 1. see: 
 https://www.youtube.com/watch?v=TkuDQ-SzXwU
I don't know if its any good but do give it a look.   For an into to python I can recommend as being very good for new programmers have a read of the following:
https://thepythoncodingbook.com

**Comment by davefes** [comment]
I think .value is for input GPIO.  Well, it is for the ESP32.

Try:
```py
while True:
    led_onboard.on()
    utime.sleep(2)
    led.onboard.off()
    utime.sleep(2)
```

**Comment by th3w** [comment]
As said, the Code in the While Part must be intendet. This issues the syntax error in the screenshot.
And If you use a Pico W then the inialization of Pin 25 must change to named Pin „LED“ instead  -> `machine.Pin(„LED“, machine.Pin.OUT)`
